pyband/transaction.py

```diff
@@ -46,22 +66,38 @@ def with_chain_id(self, chain_id: str) -> "Transaction":
         self.chain_id = chain_id
         return self
 
-    def with_fee(self, fee: int) -> "Transaction":
-        self.fee = fee
+    def with_fee(self, fee: List[Coin]) -> "Transaction":
+        self.fee = cosmos_tx_type.Fee(amount=fee, gas_limit=self.fee.gas_limit)
         return self
 
     def with_gas(self, gas: int) -> "Transaction":
-        self.gas = gas
+        self.fee.gas_limit = gas
         return self
 
     def with_memo(self, memo: str) -> "Transaction":
         if len(memo) > MAX_MEMO_CHARACTERS:
             raise ValueTooLargeError("memo is too large")
-
         self.memo = memo
         return self
 
-    def get_sign_data(self) -> bytes:
+    def __generate_info(self, publicKey: PublicKey) -> Tuple[str, str]:
+        body = cosmos_tx_type.TxBody(
+            messages=self.msgs,
+            memo=self.memo,
+        )
+        
+        any_publicKey = Any()
+        any_publicKey.Pack(publicKey.to_pubKey_proto(), type_url_prefix="")
+
+        body_bytes = body.SerializeToString()
+        mode_info = cosmos_tx_type.ModeInfo(single=cosmos_tx_type.ModeInfo.Single(mode=tx_sign.SIGN_MODE_DIRECT))
+        signer_info = cosmos_tx_type.SignerInfo(mode_info=mode_info, sequence=self.sequence, public_key=any_publicKey)
```

Comment:
signer_info = cosmos_tx_type.SignerInfo(mode_info=mode_info, sequence=self.sequence)
if public_key is not None:
  // Pack
  // Set public key